apps/analytics/activity_reports.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2016 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from datetime import datetime
+import json
+from superdesk import get_resource_service
+from superdesk.services import BaseService
+
+from eve.utils import ParsedRequest
+from superdesk.metadata.item import metadata_schema
+from superdesk.resource import Resource
+from content_api import items
+
+
+class ActivityReportResource(Resource):
+    """Activity Report schema
+    """
+
+    schema = {
+        'desk': Resource.rel('desks', nullable=True),
+        'operation': {'type': 'string'},
+        'operation_date': {'type': 'datetime'},
+        'subject': metadata_schema['subject'],
+        'keywords': metadata_schema['keywords'],
+        'group_by': {'type': 'list'},
+        'report': {'type': 'dict'},
+        'timestamp': {'type': 'datetime'},
+        'force_regenerate': {'type': 'boolean', 'default': False}
+    }
+    item_methods = ['GET', 'DELETE']
+    resource_methods = ['POST']
+
+    privileges = {'POST': 'activity_reports', 'DELETE': 'activity_reports', 'GET': 'activity_reports'}
+
+
+class ActivityReportService(BaseService):
+    # check if some fields are filled out before generating the report and initiate the filter
+    def check_for_fields(self, report):
+        terms = [
+            {"term": {"operation": report['operation']}}
+        ]
+        if report.get('subject'):
+            subjects = [subject['qcode'] for subject in report['subject']]
+            terms.append({'term': {'subject.qcode': subjects}})
+        if report.get('keywords'):
+            key = [x.lower() for x in report['keywords']]
+            terms.append({'term': {'keywords': key}})
+        if report.get('operation_date'):
+            op_date = items.ItemsService._format_date(report['operation_date'])
```

Comment:
Agreed, `content_api` is separate application based superdesk, we shouldn't use imports from `content_api` here, and function really should go to `superdesk` module, it's pretty general
```
    @staticmethod
    def _format_date(date):
        return datetime.strftime(date, ELASTIC_DATE_FORMAT)
```